sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java

```diff
@@ -182,309 +182,61 @@ private static String namespaceOf(ClassForDisplay clazz) {
-     * can be provided to point the reader to additional details about the metadata.
-     *
-     * <p>Specifying a null value will clear the URL if it was previously defined.
-     */
-    ItemBuilder withLinkUrl(@Nullable String url);
+  @AutoValue
+  public abstract static class Item<T> {
 
-    /**
-     * Adds an explicit namespace to the most-recently added display data. The namespace
-     * and key uniquely identify the display data.
-     *
-     * <p>Leaving the namespace unspecified will default to the registering instance's class.
-     */
-    ItemBuilder withNamespace(Class<?> namespace);
-
-    /**
-     * Adds an explicit namespace to the most-recently added display data. The namespace
-     * and key uniquely identify the display data.
-     *
-     * <p>Leaving the namespace unspecified will default to the registering instance's class.
-     */
-    ItemBuilder withNamespace(ClassForDisplay namespace);
-  }
-
-  /**
-   * A display data item. DisplayData items are registered via {@link Builder#add} within
-   * {@link HasDisplayData#populateDisplayData} implementations. Each metadata item is uniquely
-   * identified by the specified key and namespace generated from the registering component's
-   * class name.
-   */
-  public static class Item {
-    private final String key;
-    private final String ns;
-    private final Type type;
-    private final Object value;
-    private final Object shortValue;
-    private final String label;
-    private final String url;
-
-    private static Item create(String nsClass, String key, Type type, Object value) {
-      FormattedItemValue formatted = type.format(value);
-      return new Item(
-          nsClass, key, type, formatted.getLongValue(), formatted.getShortValue(), null, null);
-    }
-
-    private Item(
-        String namespace,
-        String key,
-        Type type,
-        Object value,
-        Object shortValue,
-        String url,
-        String label) {
-      this.ns = namespace;
-      this.key = key;
-      this.type = type;
-      this.value = value;
-      this.shortValue = shortValue;
-      this.url = url;
-      this.label = label;
-    }
-
-    @JsonGetter("namespace")
-    public String getNamespace() {
-      return ns;
-    }
+    @Nullable
+    public abstract String getNamespace();
 
     @JsonGetter("key")
-    public String getKey() {
-      return key;
-    }
+    public abstract String getKey();
 
     /**
      * Retrieve the {@link DisplayData.Type} of display data. All metadata conforms to a
      * predefined set of allowed types.
      */
     @JsonGetter("type")
-    public Type getType() {
-      return type;
-    }
+    public abstract Type getType();
 
     /**
      * Retrieve the value of the metadata item.
      */
     @JsonGetter("value")
-    public Object getValue() {
-      return value;
-    }
+    @Nullable
+    public abstract Object getValue();
```

Comment:
The value in `getValue()` is translated from the input. For example, or a `Duration` input, the value will be the count of milliseconds.

I've added additional documentation.